src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/TypeMustExistTests.Strict.cs

```diff
@@ -0,0 +1,440 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class TypeMustExistTests_Strict
+    {
+        [Theory]
+        [InlineData("")]
+        [InlineData("CP002")]
+        public void MissingPublicTypesInLeftAreReported(string noWarn)
+        {
+            string leftSyntax = @"
+
+namespace CompatTests
+{
+  public class First { }
+}
+";
+
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First { }
+  public class Second { }
+  public record MyRecord(string a, string b);
+  public struct MyStruct { }
+}
+";
+
+            ApiComparer differ = new();
+            differ.NoWarn = noWarn;
+            differ.StrictMode = true;
+            bool enableNullable = false;
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax, enableNullable);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax, enableNullable);
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(left, right);
+
+            CompatDifference[] expected = new[]
+            {
+                new CompatDifference(DiagnosticIds.TypeMustExist, $"Type 'CompatTests.Second' exists on the right but not on the left", DifferenceType.Added, "T:CompatTests.Second"),
```

Comment:
Not related to this PR but you might consider making test assertions not be so tightly coupled to resource strings.  You end up "over testing" the resource which can make maintenance harder.  Consider having a single test that checks the resource (maybe even tests the CompatDifference factory?) and then have the bulk of your tests only check the variable parts that the resource string is formatted from.

Reply:
Makes sense. I'll work on that on a separate PR to make testing looser.